crates/swift-bridge-ir/src/bridged_type/shared_struct.rs

```diff
@@ -19,18 +21,37 @@ pub(crate) struct SharedStruct {
     pub fields: StructFields,
     pub swift_name: Option<LitStr>,
     pub already_declared: bool,
+    pub is_tuple: bool,
 }
 
 impl SharedStruct {
-    pub(crate) fn swift_name_string(&self) -> String {
+    pub(crate) fn swift_name_string(
+        &self,
+        type_pos: TypePosition,
+        types: &TypeDeclarations,
+    ) -> String {
+        if self.is_tuple {
+            return self.combine_field_types_into_swift_name(type_pos, types);
+        }
         match self.swift_name.as_ref() {
             Some(ty) => ty.value(),
             None => self.name.to_string(),
         }
     }
 
-    pub(crate) fn ffi_name_string(&self) -> String {
-        let name = self.swift_name_string();
+    pub(crate) fn ffi_name_string(&self, types: &TypeDeclarations) -> String {
+        if self.is_tuple {
+            return format!(
+                "{}${}${}",
+                SWIFT_BRIDGE_PREFIX,
+                self.name.to_string(),
+                self.combine_field_types_into_ffi_name_string(types)
+            );
+        }
+        let name = match self.swift_name.as_ref() {
```

Comment:
Let's not repeat this match code for getting the swift name multiple times and instead use a re-usable method.

Reply:
We can revert these copied `match` statements if we separate `BuiltInTuple` from `SharedStruct`